packages/headless/src/features/product-listing/product-listing-actions.ts

```diff
@@ -168,17 +241,76 @@ export const buildProductListingRequest = async (
   };
 };
 
+export const buildProductListingRequestV2 = async (
+  state: StateNeededByFetchProductListingV2
+): Promise<ProductListingV2Request> => {
+  // TODO: Use sort, selected page, context
+  const facets = getFacets(state);
+  // const visitorId = await getVisitorID(state.configuration.analytics);
+
+  const baseParams = {
+    accessToken: state.configuration.accessToken,
+    organizationId: state.configuration.organizationId,
+    platformUrl: state.configuration.platformUrl,
+    version: state.productListing?.version,
+  };
+
+  const productListingParams = {
+    listingId: state.productListing.listingId,
+    locale: state.productListing.locale || 'en-US',
+    mode: state.productListing.mode || 'live',
+    clientId: state.productListing.clientId || 'some-client-id', // Dummy value since the api requires one
+  };
+
+  return {
+    ...baseParams,
+    ...productListingParams,
+    selectedFacets: facets,
+    context: {
```

Comment:
Why are we returning a bunch of dummy values? Wouldn't these be retrievable from the state object?